spec/sections/copies.bs

```diff
@@ -170,19 +175,73 @@ dictionary GPUImageCopyTexture {
     1. Return `true` if and only if all of the following conditions apply:
-        - |imageCopyTexture|.{{GPUImageCopyTexture/texture}} must be a [=valid=] {{GPUTexture}}.
-        - |imageCopyTexture|.{{GPUImageCopyTexture/mipLevel}} must be &lt;
-            |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/mipLevelCount}}.
-        - |imageCopyTexture|.{{GPUImageCopyTexture/origin}}.[=GPUOrigin3D/x=] must be a multiple of |blockWidth|.
-        - |imageCopyTexture|.{{GPUImageCopyTexture/origin}}.[=GPUOrigin3D/y=] must be a multiple of |blockHeight|.
-        - The [=imageCopyTexture physical subresource size=] of |imageCopyTexture| is equal to |copySize| if either of
-            the following conditions is true:
-            - |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/format}} is a depth-stencil format.
-            - |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/sampleCount}} &gt; 1.
+
+        <div class=validusage>
+            - |imageCopyTexture|.{{GPUImageCopyTexture/texture}} must be a [=valid=] {{GPUTexture}}.
+            - |imageCopyTexture|.{{GPUImageCopyTexture/mipLevel}} must be &lt;
+                |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/mipLevelCount}}.
+            - |imageCopyTexture|.{{GPUImageCopyTexture/origin}}.[=GPUOrigin3D/x=] must be a multiple of |blockWidth|.
+            - |imageCopyTexture|.{{GPUImageCopyTexture/origin}}.[=GPUOrigin3D/y=] must be a multiple of |blockHeight|.
+            - The [=imageCopyTexture physical subresource size=] of |imageCopyTexture| is equal to |copySize| if either of
+                the following conditions is true:
+                - |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/format}} is a depth-stencil format.
+                - |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/sampleCount}} &gt; 1.
+        </div>
+</div>
+
+<div algorithm data-timeline=device>
+    <dfn abstract-op>validating texture buffer copy</dfn>(|imageCopyTexture|, |dataLayout|, |dataLength|, |copySize|, |textureUsage|, |aligned|)
+
+    **Arguments:**
+
+    - {{GPUImageCopyTexture}} |imageCopyTexture|
+    - {{GPUImageDataLayout}} |dataLayout|
+    - {{GPUSize64Out}} |dataLength|
+    - {{GPUExtent3D}} |copySize|
+    - {{GPUTextureUsage}} |textureUsage|
+    - {{boolean}} |aligned|
+
+    **Returns:** {{boolean}}
+
+    [=Device timeline=] steps:
+
+    1. Let |texture| be |imageCopyTexture|.{{GPUImageCopyTexture/texture}}
+    1. Let |aspectSpecificFormat| = |texture|.{{GPUTexture/format}}.
+    1. Let |offsetAlignment| = [=texel block copy footprint=] of |texture|.{{GPUTexture/format}}.
+
+    1. Return `true` if and only if all of the following conditions apply:
+
+        <div class=validusage>
+            1. [$validating GPUImageCopyTexture$](|imageCopyTexture|, |copySize|) returns `true`.
+            1. [$validating texture copy range$](|imageCopyTexture|, |copySize|) return `true`.
```

Comment:
nit: It starts to look like this should be inside `validating GPUImageCopyTexture`. I think they're always together.

Reply:
Good call. Done!